prov/tcp/src/xnet_rdm.c

```diff
@@ -695,13 +695,183 @@ static struct fi_ops_ep xnet_rdm_ep_ops = {
 	.tx_size_left = fi_no_tx_size_left,
 };
 
+static int xnet_mplex_av_dup(struct util_ep *ep, struct xnet_domain *domain,
+			     struct xnet_domain *subdomain)
+{
+	int ret, i;
+	struct xnet_mplex_av *xnet_av;
+	struct util_av *util_av;
+	struct fid_av *av_fid;
+	size_t addr_size;
+	char addr[sizeof(struct sockaddr_in6)];
+	struct fi_av_attr av_attr = {
+		.type = ep->domain->av_type,
+		.count = ep->av->av_entry_pool->entry_cnt,
+		.flags = 0,
+	};
+
+	xnet_av = container_of(&ep->av->av_fid, struct xnet_mplex_av, util_av.av_fid);
+	ofi_genlock_lock(&xnet_av->lock);
+	ret = fi_av_open(&subdomain->util_domain.domain_fid, &av_attr, &av_fid, NULL);
+	if (ret)
+		goto out;
+	util_av = container_of(av_fid, struct util_av, av_fid);
+	for (i = 0; i < ep->av->av_entry_pool->entry_cnt; i++) {
+		ret = fi_av_lookup(&ep->av->av_fid, i, addr, &addr_size);
+		if (ret)
+			continue;
+		ret = ofi_av_insert_addr_at(util_av, addr, i);
+		if (ret)
+			goto out;
+	}
```

Comment:
This loop goes into the AV implementation.  It should use AV calls directly, even if internal AV calls.  If the AV needs to adjust to handle an invalid fi_addr_t index value, it should do that.